### PR TITLE
Fix snapping to the wrong diagonal axis

### PR DESCRIPTION
**Pull request by tycrek**
Added a check for player Yaw being outside the valid range (0 to 180 and -0 to -180). There might be a bit nicer way to do this but this worked for me.

The issue I was facing is that sometimes on diagonals, the bot would snap to a different axis than the one the player was facing. For example I noticed that Forge was reporting Yaw as `-225` instead of `135`. After adding the fix I had no issues on any axis.

Built and tested using IntelliJ; latest Forge for Minecraft 1.12.2

### CODE REVIEW

**Comment by tycrek** [comment]
This fix might also be needed on the "normal" axis. I haven't actually tested those out but it should be easy enough to copy the code.

**Comment by bebeli555** [comment]
Thank you and sorry that i didn't credit you properly for this.
